Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,937 @@
+            break;
+          if ($(dropDownMenuItems[currentItemIndex]).children('a').length > 0) // skip separators
+            break;
+        }
+        break;
+      case 38: // up arrow
+        event.preventDefault();
+        event.stopPropagation();
+        isSelectionUpdated = true;
+        oldIndex = currentItemIndex >= 0 ? currentItemIndex : itemInfos.length - 1;
+        while (true) // skip separators
+        {
+          if (currentItemIndex > 0)
+            currentItemIndex--;
+          else
+            currentItemIndex = itemInfos.length - 1;
+
+          if (oldIndex === currentItemIndex)
+            break;
+          if ($(dropDownMenuItems[currentItemIndex]).children('a').length > 0)
+            break;
+        }
+        break;
+    }
+
+    if (isSelectionUpdated)
+    {
+      $("li", DropDownMenu._currentPopup).removeClass (DropDownMenu._itemSelectedClassName);
+      if (currentItemIndex >= 0 && currentItemIndex < itemInfos.length)
+      {
+        var dropDownMenuItem = $(dropDownMenuItems[currentItemIndex]);
+        if (dropDownMenuItem.children('a').length > 0) // skip separators
+        {
+          dropDownMenuItem.addClass(DropDownMenu._itemSelectedClassName);
+          var nonJQueryAnchor = dropDownMenuItem.children ('a')[0];
+          // do not use jQuery focus() here because jQuery causes the focus/blur events to fire in the wrong order (first focus, then blur)
+          nonJQueryAnchor.focus();
+        }
+      }
+    }
+  }
+
+  private static GetItemEnabled (itemInfo: DropDownMenu_ItemInfo, selectionCount: number): boolean
+  {
+    var isEnabled = true;
+    if (itemInfo.IsDisabled)
+    {
+      isEnabled = false;
+    }
+    else
+    {
+      if (itemInfo.RequiredSelection == DropDownMenu._requiredSelectionExactlyOne
+          && selectionCount != 1)
+      {
+        isEnabled = false;
+      }
+      if (itemInfo.RequiredSelection == DropDownMenu._requiredSelectionOneOrMore
+          && selectionCount < 1)
+      {
+        isEnabled = false;
+      }
+    }
+    return isEnabled;
+  }
+
+  private static GetTarget (event: JQueryMouseEventObject, tagName: string): Element
+  {
+    var element: Nullable<Element> = event.target;
+    while (element && element.tagName != tagName)
+      element = element.parentNode as Nullable<Element>;
+    // more fun with IE, sometimes event.target is empty, just ignore it then
+    if (!element)
+      return [] as unknown as Element;
+    return element;
+  }
+
+  private static FocusHandler (): void
+  {
+    if (DropDownMenu._blurTimer !== null)
+      clearTimeout(DropDownMenu._blurTimer);
+  }
+
+  private static BlurHandler (): void
+  {
+    if (DropDownMenu._blurTimer !== null)
+      clearTimeout (DropDownMenu._blurTimer);
+
+    DropDownMenu._blurTimer = setTimeout (function () { DropDownMenu.ClosePopUp (!DropDownMenu._updateFocus); }, 50);
+  }
+
+  private static ItemClickHandler (this: HTMLAnchorElement): boolean
```

Comment:
Move to after `OnHeadControlClick`

Consider renaming to `OnItemClick`

Since we a) move the method and b) changed the exact name anyway, we can also fix the name. And yes, I'm aware that there's a bunch of different conventions going on in this file.